### ISSUE TITLE
fix(compiler,vm): decode OpMethod's kind operand as a typed enum

### ISSUE THREAD

**Issue by gengjiawen** [gengjiawen/monkey-rust#321]
`OpMethod`'s second operand is a `MethodKind` discriminant. It is hand-encoded on one side and hand-decoded on two others, with nothing tying the three together.

**Encode** — `compiler/compiler.rs:386`:

```rust
let kind = match method.kind {
    MethodKind::Method => 0,
    MethodKind::Constructor => 1,
};
self.emit_with_span(OpMethod, &[method_name, kind], &method.span);
```

**Decode, classic VM** — `compiler/vm.rs:305`:

```rust
if kind == 1 { class.borrow_mut().constructor = Some(method); }
else         { class.borrow_mut().methods.insert(name, method); }
```

**Decode, GC VM** — `gc/vm.rs:588`:

```rust
let result = self.install_method(class, name, method, kind == 1);
```

`MethodKind` has two variants today (`parser/ast.rs:85`), so compiler-generated bytecode does not trigger the forward-evolution miscompile yet. The decoder bug already exists for untrusted snapshots, though: any value other than `1` currently passes validation and is treated as an ordinary method.

Add a third variant — say `MethodKind::Getter` — and valid compiler output reaches the same bug. The compiler side is a `match`, so rustc forces you to handle the new AST variant. Both VM sides are a boolean test on `kind == 1`, so they compile unchanged and silently file every getter as an ordinary method. Same failure shape as the operand truncation #317 fixed: compiles, runs, wrong behaviour — reached by a different route, and now in two places instead of one.

## A range check does not fix this

Worth stating explicitly, because it was the wrong answer in an earlier draft of #320: describing this operand in the opcode table as a bounded tag (`Tag { variants: N }`) only lets the encoder reject values outside `0..N`. The failure here is an **in-range** value being misinterpreted. Bumping `variants` to 3 makes the encoder accept `2` and changes nothing about `kind == 1`.

## Proposal

A shared wire enum that both ends must go through:

```rust
#[repr(u8)]
#[derive(Clone, Copy, PartialEq)]
pub enum WireMethodKind {
    Method = 0,
    Constructor = 1,
}

impl TryFrom<u8> for WireMethodKind { /* reject unknown discriminants */ }
```

- keep the wire enum in the bytecode/compiler layer rather than giving the parser AST enum a wire representation
- convert AST `MethodKind` to `WireMethodKind` through an exhaustive `From` implementation or local `match`, then encode the wire enum as `u8`; this keeps rustc's exhaustiveness check without coupling AST variant order to the ABI
- both VMs decode via `try_from` and then `match` **exhaustively**, so a new wire variant is a compile error in every consumer rather than a silent fallthrough
- `validate_instruction_stream` (`compiler/snapshot.rs:429`) rejects unknown discriminants on load

That last point is a real gap independent of the rest: the validator already bounds-checks `OpConst`'s constant index, `OpClosure`'s capture list and jump targets, but does not look at `OpMethod`'s kind at all. `.mbc` input is untrusted (see the comment at `snapshot.rs:268`), so an out-of-range kind currently sails through validation and lands in `kind == 1`.

## Snapshot / ABI obligation

`compute_abi_fingerprint()` (`compiler/snapshot.rs:86`) absorbs opcode values, names and operand *widths*. Adding `Getter = 2` changes no width, so the fingerprint is unchanged and an older VM would accept a newer `.mbc` and then misdecode it.

Whichever way this is fixed, the rule should be written down: **when a tag set or a discriminant mapping changes, either a stable semantic identifier goes into the fingerprint or `FORMAT_VERSION` moves.** Same applies to any future tag-shaped operand.

## Related

- #320 — operand metadata in the opcode table; this issue is the part a spec cannot solve
- #319 — compile errors carry no source location
